synapse/lib/layer.py

```diff
@@ -3352,6 +3352,67 @@ async def setStorNodeProp(self, buid, prop, valu, meta):
         _, changes = await self.saveNodeEdits(nodeedits, meta)
         return bool(changes[0][2])
 
+    async def delStorNode(self, buid, meta):
+        '''
+        Delete all node information in this layer.
+
+        Deletes props, tagprops, tags, n1edges, n2edges, nodedata, and node valu.
+        '''
+        sode = self._getStorNode(buid)
+        if sode is None:
+            return False
+
+        formname = sode.get('form')
+
+        edits = []
+        for propname, propvalu in sode.get('props', {}).items():
+            edits.append(
+                (EDIT_PROP_DEL, (propname, *propvalu), ())
+            )
+
+        for tagname, tprops in sode.get('tagprops', {}).items():
+            for propname, propvalu in tprops.items():
+                edits.append(
+                    (EDIT_TAGPROP_DEL, (tagname, propname, *propvalu), ())
+                )
+
+        for tagname, tagvalu in sode.get('tags', {}).items():
+            edits.append(
+                (EDIT_TAG_DEL, (tagname, tagvalu), ())
+            )
+
+        async for edge in self.iterNodeEdgesN1(buid):
+            edits.append(
+                (EDIT_EDGE_DEL, edge, ())
+            )
+            await asyncio.sleep(0)
+
+        async for item in self.iterNodeData(buid):
```

Comment:
If `valu` is present in the node, we could skip nodedata edits since `_editNodeDel` wipes node data.